client/client.go

```diff
@@ -2294,13 +2296,22 @@ func (client *OpenFgaClient) singleBatchCheck(ctx _context.Context, body fgaSdk.
 	}
 
 	requestID := fmt.Sprintf("%d", time.Now().UnixNano())
+	headerName := "X-OpenFGA-Client-Bulk-Request-Id"
 
-	client.APIClient.GetConfig().AddDefaultHeader("X-OpenFGA-Client-Bulk-Request-Id", requestID)
+	// Create a mutex-protected copy of headers to avoid race conditions
+	client.headerMutex.Lock()
+	client.APIClient.GetConfig().AddDefaultHeader(headerName, requestID)
 
 	req := client.OpenFgaApi.BatchCheck(ctx, *storeId)
 	req = req.Body(body)
 
 	response, _, err := req.Execute()
+
+	// Remove the header after use to clean up
+	client.APIClient.GetConfig().DefaultHeaders[headerName] = ""
+	delete(client.APIClient.GetConfig().DefaultHeaders, headerName)
```

Comment:
do we need this because otherwise all future requests would have the `X-OpenFGA-Client-Bulk-Request-Id` header? Do we not have a way to add a header just to this specific request?

Reply:
Yes, the default header would affect future requests otherwise.  I'm not seeing a straightforward way to apply it on a per-request basis but would be open to it if someone else more knowledgable does.